flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/sql/JoinTest.scala

```diff
@@ -55,12 +55,13 @@ class JoinTest extends TableTestBase {
           unaryNode(
             "DataStreamCalc",
             streamTableNode(1),
-            term("select", "a", "b", "proctime")
+            term("select", "a", "b", "-(proctime, 3600000) AS -",
```

Comment:
I think it's easy to prevent the time indicator push-down in our side. Calcite 1.3 changed the `ProjectJoinTransposeRule.INSTANCE` to `new ProjectJoinTransposeRule(         PushProjector.ExprCondition.TRUE, RelFactories.LOGICAL_BUILDER);` instead of `new ProjectJoinTransposeRule(PushProjector.ExprCondition.FALSE, RelFactories.LOGICAL_BUILDER);`.  

In order to change the default behavior of the rule, the only thing we need to do is to create a new `ProjectJoinTransposeRule` with a custom `PushProjector.ExprCondition` which filters prevent time indicator rex nodes.